contracts/hyperlane-interfaces/src/lib.sw

```diff
@@ -4,6 +4,21 @@ dep igp;
 
 use hyperlane_message::Message;
 
+/// Logged when a message is dispatched.
+/// Although the message ID can be calculated by hashing the message contents,
+/// this is also logged for convenience.
+pub struct DispatchIdEvent {
+    message_id: b256,
+}
+
+/// Logged when a message is processed.
+pub struct ProcessEvent {
```

Comment:
any reason not to mirror the solidity events here?

Reply:
The only reason we separated the solidity events was to have a bunch of indexed params, since that's not a thing in Fuel I figured no reason to do that and unnecessarily have separate events?

Reply:
I don't feel super strongly here, maybe deciding whether this is something we want to do can be done when refining things before audit?

Reply:
We're already forced into some inconsistency because we are logging the message bytes by themselves, whereas in solidity we have
```
    event Dispatch(
        address indexed sender,
        uint32 indexed destination,
        bytes32 indexed recipient,
        bytes message
    );
```